### ISSUE TITLE
`flat` is not consistent with `Array.flat`

### ISSUE THREAD

**Issue by jasonk** [vitaly-t/iter-ops#228]
The docs say that the flat operator is consistent with `Array.flat`, but then show an example that is not consistent, as `Array.flat` leaves non-array values alone and would not expand `one` into separate characters.

```javascript
console.log( ...pipe( [ 'one', [ 2, 3, [ 4, 5 ] ] ], flat( 2 ) ) );
```
> o n e 2 3 4 5

```javascript
console.log( ...[ 'one', [ 2, 3, [ 4, 5 ] ] ].flat( 2 ) );
```
> one 2 3 4 5

**Comment by vitaly-t** [comment]
That's true, `Array.flat`, as it turns out, doesn't really handle generic iterables like this library does, it only handles arrays. A string is an iterable object, that's why this library flattens it according to the depth/level specified.

A correct description should be that `flat` extends `Array.flat`, by applying `flat` to all iterable objects inside the list.

**Comment by vitaly-t** [comment]
Following this, I have added the [following idea](https://github.com/vitaly-t/iter-ops/discussions/229), as a possible extension.

**Comment by vitaly-t** [comment]
This has been addressed in version [3.2.0](https://github.com/vitaly-t/iter-ops/releases/tag/3.2.0), by adding optional `skip` callback, and updating [documentation](https://vitaly-t.github.io/iter-ops/functions/flat.html).

**Comment by vitaly-t** [comment]
In addition to this, I have added [flat/flatMap](https://github.com/vitaly-t/iter-ops-extras/blob/main/src/flat.ts) overrides within the [extras](https://github.com/vitaly-t/iter-ops-extras), in case those are preferred.
